query-node/schemas/forum.graphql

```diff
@@ -20,6 +20,7 @@ type ForumCategory @entity {
   id: ID!
 
   "Parent category (if none - this is a root category)"
+  parentId: ID
```

Comment:
Is this field added for doing nested filtering, if yes please mark this field with the [comment](https://github.com/thesan/joystream/blob/cf06f473330110c157e5c18bc2886f3a00428b58/query-node/schemas/contentEvents.graphql#L24), it will help to remove all the redundant fields once nested filtering is available.

Reply:
I think I got confused here :zany_face:. This is only used in the mappings [canModerate](https://github.com/Joystream/joystream/blob/cf06f473330110c157e5c18bc2886f3a00428b58/query-node/mappings/src/forum.ts#L89) so there is no need to add this to the schema

Reply:
Never mind these needs to be exposed in schema apparently... my understanding of hydra is very limited!
So I just added nicer comments in the schema but solving nested filtering won't affect these.

Reply:
That's what I was thinking last Friday but it looks like the `ForumCategory` type in `query-node/generated/graphql-server/dist/src/modules/forum-category/forum-category.model.d.ts` is generated from the schema.

Here are the type errors: https://github.com/Joystream/joystream/actions/runs/3252154043/jobs/5338037876

The CI reported on the commit with this changes: https://github.com/Joystream/joystream/pull/4317/commits/ba8a8e75013e8220efe4c6f6ea256a4216325218#diff-0edeabbbad7f1c259beb22e445370523d09b886b184534cdd8e8bd1ffdb5c448

Or is there another way to access these ids ?

Reply:
> Or is there another way to access these ids?

I believe you are talking about this piece of code, if yes

```ts
async function canModerate(store: DatabaseManager, actorId: string, categoryId: string): Promise<boolean> {
  const category = await getCategory(store, categoryId, ['moderators'])
  return (
    category.moderators.some(({ id }) => id === actorId) ||
    (category.parentId ? canModerate(store, actorId, category.parentId) : false)
  )
}
```
Then you can actually get the id using parent relation, i.e., `parent.id`